system/jlib/jutil.cpp

```diff
@@ -2437,22 +2457,46 @@ jlib_decl bool querySecuritySettings(bool *          _useSSL,
         if (_useSSL)
             *_useSSL = useSSL;
         if (_port)
-            *_port = useSSL ? SECURE_DAFILESRV_PORT : DAFILESRV_PORT;
+        {
+            // if we must use ssl port ...
+            if (useSSL == SSLOnly)
+                *_port = SECURE_DAFILESRV_PORT;
+            else
+                *_port = DAFILESRV_PORT;
+        }
         if (_certificate)
             *_certificate = certificate.get();
         if (_privateKey)
             *_privateKey = privateKey.get();
+        if (_passPhrase)
+            *_passPhrase = passPhrase.get();
     }
     else
     {
         if (_useSSL)
-            *_useSSL = false;
+            *_useSSL = SSLNone;
         if (_port)
             *_port = DAFILESRV_PORT;
     }
     return retrieved;
 }
 
+jlib_decl bool queryDafsServerSecSettings(SSLCfg *        _useSSL,
+                                          unsigned short *_port,
+                                          unsigned short *_sslport,
+                                          const char * *  _certificate,
+                                          const char * *  _privateKey,
+                                          const char * *  _passPhrase)
+{
+    bool ret = querySecuritySettings(_useSSL, nullptr, _certificate, _privateKey, _passPhrase);
+    // these should really be in env, but currently they are not ...
+    if (_port)
+        *_port = DAFILESRV_PORT;
```

Comment:
bit odd that querySecuritySettings if SSLonly would/could have filled this in..

would be clearer if unified method.